.github/workflows/reusable-build-docsite.yaml

```
@@ -0,0 +1,28 @@
+name: Build the Ansible docsite
+
+on:
+  workflow_call:
+
+jobs:
+  build-docsite:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repository
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v3
+        with:
+          python-version: "3.10"
+      - name: Install requirements
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install -r requirements.in -c requirements.txt
+      - name: Generate assets for the docs landing page
+        run: python build.py
```

Here's how you can catch warnings like the ones about unclosed resources (as in file descriptors), erroring out early:
```suggestion
        run: python -I -Werror build.py
```